packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -204,8 +2129,47 @@ class InputDecoration {
   /// If null, defaults to the [helperStyle].
   final TextStyle counterStyle;
 
-  /// Creates a copy of this input decoration but with the given fields replaced
-  /// with the new values.
+  /// If true the decoration's container is filled with [fillColor].
+  ///
+  /// Typically this field set to true if [borderType] is
+  /// [InputBorderType.underline].
+  ///
+  /// The decoration's container is the area which is filled if [isFilled] is
+  /// true and bordered per the [borderType]. It's the area adjacent to
+  /// [decoration.icon] and above the widgets that contain [helperText],
+  /// [errorText], and [counterText].
+  ///
+  /// This property is false by default.
+  final bool filled;
```

Comment:
should we just fill if fillColor is non-null, and not fill if it's null?

Reply:
Most apps will not want to actually specify a fill color, they'll just the default one that's based on the theme's brightness.

Reply:
(that would work better in a world where there's a way to configure the fill color in the theme explicitly)

Reply:
I agree.  What might work best here is an InputDecoratorTheme which would be part of ThemeData like IconTheme and textTheme. I will add that in a follow-up PR.